### PR TITLE
Fix password saving when password is entered while connecting. 

### LINKED ISSUE #475: Password not saved when savePassword is enabled and user is prompted for password at time of connection
- MSSQL Extension Version:  0.1.5
- VSCode Version: 1.7.2
- OS Version: Windows 10

Steps to Reproduce:

1. Paste profile into mssql.connections:
        {
            "server": "Server Name",
            "database": "",
            "authenticationType": "SqlLogin",
            "user": "Username",
            "password": "",
            "savePassword": true,
            "profileName": "Test Profile"
        }
2. Connect to profile and enter password
3. Disconnect
4. Connect to profile again 
**Notice how password is not saved**

### PR DESCRIPTION
**Pull request by Raymondd**
Fixes #475 Saves password when password is entered while connecting.  Also changed a line from my last pull request that wasn't needed. The tests I wrote on my last pull request should cover these changes.

### CODE REVIEW

File: src/models/connectionCredentials.ts [hunk 1]
```diff
@@ -118,13 +118,14 @@ export class ConnectionCredentials implements IConnectionCredentials {
...
+                        connectionStore.removeProfile(profile).then(() => {
                             connectionStore.saveProfile(profile);
                         });
                     // Or, if the user answered any additional questions for the profile, be sure to save it
                     } else if (profile.authenticationType !== unprocessedCredentials.authenticationType ||
-                               profile.savePassword !== (<IConnectionProfile>unprocessedCredentials).savePassword) {
+                               profile.savePassword !== (<IConnectionProfile>unprocessedCredentials).savePassword ||
+                               profile.password !== unprocessedCredentials.password) {
                         connectionStore.removeProfile(profile).then(() => {
                             connectionStore.saveProfile(profile, !wasPasswordEmptyInConfigFile);
```
Comment (kevcunnane) [inline src/models/connectionCredentials.ts]: >wasPasswordEmptyInConfigFile [](start = 66, length = 28)

Please make this explicitly false.

File: src/models/connectionCredentials.ts [hunk 2]
```diff
@@ -118,13 +118,14 @@ export class ConnectionCredentials implements IConnectionCredentials {
...
-                        // Remove profile, but keep credential store if savePassword was enabled
-                        connectionStore.removeProfile(profile, true).then(() => {
+                        // Remove profile, then save profile without plain text password
+                        connectionStore.removeProfile(profile).then(() => {
                             connectionStore.saveProfile(profile);
                         });
                     // Or, if the user answered any additional questions for the profile, be sure to save it
                     } else if (profile.authenticationType !== unprocessedCredentials.authenticationType ||
-                               profile.savePassword !== (<IConnectionProfile>unprocessedCredentials).savePassword) {
+                               profile.savePassword !== (<IConnectionProfile>unprocessedCredentials).savePassword ||
```
Comment (kevcunnane) [inline src/models/connectionCredentials.ts]: Please add unit tests to cover the different scenarios here.

**Comment by kevcunnane** [comment]
:clock1:

**Comment by Raymondd** [comment]
The issue blocking Linux builds had to do with capitalization sensitivity. It couldn't find an import due a capital letter, seems like OSX and Windows didn't have the same problem though. Just for future reference.

**Pull request merged**